### ISSUE TITLE
Real number constraints not respected correctly

### ISSUE THREAD

**Issue by dionny** [chocoteam/choco-solver#415]
I am attempting to model the following constraints over real numbers with a precision of `0.1`:
`50.00 <= min <= 100.00`
`50.00 <= max <= 100.00`
`min <= max`

I've used Choco as follows (alongside with ibex-lib):
```
double precision = 1.0e-1;
RealVar min = VariableFactory.real("min", Double.NEGATIVE_INFINITY, Double.POSITIVE_INFINITY, precision, solver);
RealVar max = VariableFactory.real("max", Double.NEGATIVE_INFINITY, Double.POSITIVE_INFINITY, precision, solver);

vars = new RealVar[]{min, max};

// min <= max
solver.post(new RealConstraint(UUID.randomUUID().toString(), "{0} <= {1}", Ibex.HC4, vars));

// 50.00 <= min <= 100.00
solver.post(new RealConstraint(UUID.randomUUID().toString(), "{0} >= 50.00", Ibex.HC4, vars));
solver.post(new RealConstraint(UUID.randomUUID().toString(), "{0} <= 100.00", Ibex.HC4, vars));

// 50.00 <= max <= 100.00
solver.post(new RealConstraint(UUID.randomUUID().toString(), "{1} >= 50.00", Ibex.HC4, vars));
solver.post(new RealConstraint(UUID.randomUUID().toString(), "{1} <= 100.00", Ibex.HC4, vars));

Chatterbox.showSolutions(solver);

solver.set(new RealStrategy(vars, new Cyclic(), new RealDomainMiddle()));

solver.findSolution();
solver.getIbex().release();
```
And I get back the following solution:
```
Solution #1 found. Solver[Simple], 1 Solutions, Resolution time 0.002s, 19 Nodes (9,285.5 n/s), 0 Backtracks, 0 Fails, 0 Restarts 
	min = [50.0000000000000000,50.0976562500000000] max = [50.0000000000000000,50.0976562500000000] .
```

**The problem:**
Everything looks OK; however, when I change one of the constraints (from min <= max, to min < max), I get back the same exact solution, which doesn't make sense.

Because of this, a number of questions come up:
Am I using Choco incorrectly for the problem I am trying to model? 
Am I reading the solutions incorrectly? 
What is the correct way to get back a specific real number for each of my variables, rather than a range?

**Comment by cprudhom** [comment]
I reproduce this issue. I let you know when it comes from

**Comment by cprudhom** [comment]
Ok, I've just got answers from Gilles Chabert (software architecte and main developer of Ibex).
He told me that, most of the time (not to say always) continuous calculus libraries do not handle open intervals, since, due to rounding issues, it will fail to maintain correctly this information.
And, operators "<" and "<=" act similarly and are just there for syntactic sugar. In very rare cases, domains can be contracted wisely. Otherwise, you simply trying to adapt discrete concepts to continuous world.

Plus, in Choco, we have the notion of *precision* that indicates when a variable is instantiated, but such notion does not exist in Ibex. 

In other words, the two operators are handled the same way in Ibex, (and thus in Choco with RealVariables) and we cannot say that there is an issue.
